### DISCUSSION TITLE
How to update STM32duino core from 1.9 to >2.0

### DISCUSSION THREAD

**Discussion by dungar43** [stm32duino/Arduino_Core_STM32#1974] [Libraries]
I've been reading the wiki and blogs but I cant figure out how to update STM32duino core from 1.9 to >2.0. Im getting this error
on Arduino ide 2.0 " Due to API change, this sketch is compatible with STM32_CORE_VERSION  >= 0x02000000".

**Comment by fpistm** [comment]
Uninstall 1.9.0 then change the package index url available in the wiki.

**Comment by dungar43** [comment]
Thanks for the reply. I'm a newcomer to this and dont know how to uninstall 1.9.0.  Do I use the boards manager on arduino-ide 2. I've searched for STM32duino in my file manager and cant find it.

**Reply by fpistm** [reply]
As stated by the 2.0.0 release:
https://github.com/stm32duino/Arduino_Core_STM32/releases/tag/2.0.0

Board package index has changed. So in the preferences you probably have the old link:
https://raw.githubusercontent.com/stm32duino/BoardManagerFiles/master/STM32/package_stm_index.json
or 
https://raw.githubusercontent.com/stm32duino/BoardManagerFiles/master/package_stmicroelectronics_index.json

So remove uninstall it thanks the Arduino Board manager. Select contributed then you should see the installed core and click on the `Installed` button which will change to `uninstall`.

Then change in the preference the json file:
https://github.com/stm32duino/BoardManagerFiles/raw/main/package_stmicroelectronics_index.json

If you have any issue, probably the IDE doesn't refresh properly the index file. In that case you could try to remove it manually in the installation folder:
`c:\Users\<username>\AppData\Local\Arduino15\`

**Comment by dungar43** [comment]
Again thanks for the help. This is what I see in the Boards Manager
[image: image.png] Is the second item the Arduino_Core. I uninstalled and
instale\led it to no effect.

**Reply by fpistm** [reply]
No image.

**Comment by dungar43** [comment]
See the attachment

**Reply by fpistm** [reply]
> See the attachment

There is no attachment. It is not a mail.

**Comment by dungar43** [comment]
When I list the "Contributed" in Boards Manage in Arduino IDE 2.0  I see:

   1. Industrino...Install
   2. STM32 MCU based boards by STMicroelectronics Version 2.3.0  Installed
   3. STM8 MCU based boards by   STMicroelectronics
   Install
   4. Windows 10 IOT Core
                       Install

Is #2 the correct one? By the way, I can run the same code on ARDUIN-IDE 1.9
without a problem. Is there something different about Arduino-IDE 2.0 ?

**Comment by fpistm** [comment]
'#2  is the correct one. You should be able to update

**Comment by dungar43** [comment]
Thanks, I'll check it out.
